### PR TITLE
Adds exports for types

### PR DESCRIPTION
**Pull request by dbrudner**
We discovered when creating types for react-recurly that our current types for recurly-js aren't exporting types. While the current types are sufficient for declaring Recurly, if a consumer needs to import a specific type right now, they would be unable to do so.

This change removes the declaration generation process, which is unecessary, in favor of writing our own declaration file that declares Recurly and also exports types from the lib.

Example usage (see test/types/recurly-recurly-error.ts):

```
import { RecurlyError } from "@recurly/recurly-js"

function myRecurlyErrorHandler (error: RecurlyError) {
    ...
}
```

### CODE REVIEW

File: test/types/recurly-error.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { RecurlyError } from '../../types';
```
Comment (dbrudner) [inline test/types/recurly-error.ts:1]: This test shows what I'm trying to achieve. With this change, we can pull `RecurlyError` and other types out the definition and use them as needed.

**Comment by chrissrogers** [comment]
Thinking we should export all types instead of a subset, to ensure broad access -- if a lib using these types as a dep needs something, then it will be available to them.

File: Makefile [hunk 2]
```diff
@@ -68,5 +62,5 @@ clean:
 	@rm -rf node_modules build
 
 .PHONY: server server-http
-.PHONY: test-ci test-unit test-unit-ci test-unit-cov-ci test-e2e test-e2e-ci test-types
```
Comment (chrissrogers) [inline Makefile:71]: This needs to remain 💥

File: test/types/tsconfig.json [hunk 3]
```diff
@@ -6,6 +6,10 @@
     "strictNullChecks": false,
     "strictFunctionTypes": false,
     "esModuleInterop": true,
-    "types": []
+    "types": [],
+    "baseUrl": "./",
+    "paths": {
+      "@recurly/recurly-js": ["./index.d.ts"]
```
Comment (chrissrogers) [inline test/types/tsconfig.json:12]: is this essential?

File: types/3d-secure.ts [hunk 4]
```diff
@@ -14,4 +14,4 @@ type ThreeDSecure = (riskOptions: RiskOptions) => ThreeDSecureEmitter;
 
 type Risk = () => { ThreeDSecure: ThreeDSecure };
 
-export default Risk;
+export { Risk };
```
Comment (chrissrogers) [inline types/3d-secure.ts:17]: nit: `export type Risk`

File: types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+export * from './apple-pay';
+export * from './bank-account';
+export * from './configure';
+export * from './elements';
+export * from './error';
+export * from './gift-card';
+export * from './paypal';
+export * from './recurly';
+export * from './token';
+export * from './validate';
```
Comment (chrissrogers) [inline types/index.d.ts:31]: This is awesome.

File: test/types/tslint.json [hunk 6]
```diff
@@ -1,6 +0,0 @@
-{
```
Comment (dbrudner) [inline test/types/tslint.json:1]: We no longer need this as we're not violating any dtslint rules.

**Pull request merged**